test/ref_ops_test.cpp

```diff
@@ -2674,6 +2674,165 @@ TEST_CASE(gathernd_test)
+
+        std::vector<float> data_vec(2 * 3 * 1);
+        std::iota(data_vec.begin(), data_vec.end(), 0);
+        std::vector<int64_t> indices_vec{1, 0, 0, 1};
+
+        auto xdata  = mm->add_parameter("X", ds);
+        auto xindex = mm->add_parameter("I", is);
+
+        auto gathernd_op = migraphx::make_op("gathernd");
+        auto gathernd    = mm->add_instruction(gathernd_op, xdata, xindex);
+
+        mm->add_return({gathernd});
+        p.compile(migraphx::ref::target{});
+
+        migraphx::parameter_map params;
+        migraphx::shape input_fixed_shape0{migraphx::shape::float_type, {2, 3, 1}}; // data
+        migraphx::shape input_fixed_shape1{migraphx::shape::int64_type, {2, 2, 1}}; // index
+
+        params["X"] = migraphx::argument(input_fixed_shape0, data_vec.data());
+        params["I"] = migraphx::argument(input_fixed_shape1, indices_vec.data());
+
+        auto result = p.eval(params).back();
+        std::vector<float> res_data{};
+        std::vector<float> gold{3, 4, 5, 0, 1, 2, 0, 1, 2, 3, 4, 5};
+        result.visit([&](auto output) { res_data.assign(output.begin(), output.end()); });
+
+        EXPECT(migraphx::verify_range(res_data, gold));
+    }
+
+    { // dynamic data, dims not fixed
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+
+        migraphx::shape ds{migraphx::shape::float_type, {{2, 5, 2}, {1, 5, 0}, {1, 5, 0}}};
+        migraphx::shape is{migraphx::shape::int64_type, {2, 2, 1}};
+
+        std::vector<float> data_vec(2 * 3 * 1);
+        std::iota(data_vec.begin(), data_vec.end(), 0);
+        std::vector<int64_t> indices_vec{1, 0, 0, 1};
+
+        auto xdata  = mm->add_parameter("X", ds);
+        auto xindex = mm->add_parameter("I", is);
+
+        auto gathernd_op = migraphx::make_op("gathernd");
+        auto gathernd    = mm->add_instruction(gathernd_op, xdata, xindex);
+
+        mm->add_return({gathernd});
+        p.compile(migraphx::ref::target{});
+
+        migraphx::parameter_map params;
+        migraphx::shape input_fixed_shape0{migraphx::shape::float_type, {2, 3, 1}}; // data
+        migraphx::shape input_fixed_shape1{migraphx::shape::int64_type, {2, 2, 1}}; // index
+
+        params["X"] = migraphx::argument(input_fixed_shape0, data_vec.data());
+        params["I"] = migraphx::argument(input_fixed_shape1, indices_vec.data());
+
+        auto result = p.eval(params).back();
+        std::vector<float> res_data{};
+        std::vector<float> gold{3, 4, 5, 0, 1, 2, 0, 1, 2, 3, 4, 5};
+        result.visit([&](auto output) { res_data.assign(output.begin(), output.end()); });
+
+        EXPECT(migraphx::verify_range(res_data, gold));
+    }
+
+    { // dynamic both index and data
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+
+        migraphx::shape ds{migraphx::shape::float_type, {{2, 5, 2}, {1, 5, 0}, {1, 5, 0}}};
+        migraphx::shape is{migraphx::shape::int64_type, {{2, 5, 3}, {2, 2, 3}, {1, 1}}};
```

Comment:
{2, 2, 3} is not a valid `dynamic_dimension`

Reply:
It isn't supposed to be, but the existing code allows it.  I've included dimension specifications in this test that initialize in different ways.  Why not include tests of what should be invalid input?

Reply:
This won't error out though. If you wanted to test this or make a new assert to prevent this I would make a new PR doing this for the `shape` class and related tests rather than in `gathernd`.

Reply:
changed, we agreed this op. is not the place to check for invalid dynamic dimension inputs